lib/src/gradle_parser.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:petitparser/petitparser.dart';
+
+/// Grammar for Gradle
+class GradleGrammerDefinition extends GrammarDefinition {
+  @override
+  Parser start() => ref0(value).end();
+
+  /// A token, smallest unit to parse
+  Parser token(Object source, [String name]) {
+    if (source is String) {
+      return source.toParser(message: 'Expected ${name ?? source}').trim();
+    } else if (source is Parser) {
+      ArgumentError.checkNotNull(name, 'name');
+      return source.flatten('Expected $name').trim();
+    } else {
+      throw ArgumentError('Unknown token type: $source.');
+    }
+  }
+
+  /// Prodctions to parse a block of elements
+  Parser block() => ref1(token, '{') & ref0(elements) & ref1(token, '}');
+
+  /// A list of key-value pairs seperated by a new line
+  Parser elements() =>
+      ref0(pair).separatedBy(ref1(token, '\n'), includeSeparators: false);
+
+  /// Proudctions to parse each block
+  Parser members() =>
+      ref0(pair).separatedBy(ref1(token, '\n'), includeSeparators: false);
+
+  /// a key-value pair seperated by a whitespace
+  Parser pair() => ref0(stringToken) & ref1(token, ' ') & ref0(value);
+
+  /// Primitive value tokens
+  Parser trueToken() => ref1(token, 'true');
+  Parser falseToken() => ref1(token, 'false');
+  Parser stringToken() => ref2(token, ref0(stringPrimitive), 'string');
+  Parser numberToken() => ref2(token, ref0(numberPrimitive), 'number');
+
+  /// Possible values for parsing
+  Parser value() => [
```

Comment:
You might need to reference gradle documentation here... find the grammar and reference the rules...

Or we need to find a way to recover from errors...

Reply:
Okay, I could not find the perfect docs for this, just referred things here and there to find the structure specific to android, maybe I'll have to dig in a bit deeper.

Reply:
Yeah, the stuff we talked about in [hackers-pub-dev-](https://github.com/flutter/flutter/wiki/Chat) is similar in many ways... Tokenize the input, that'll strip comments and ideally also handle strings (so we ignore brackets inside strings), then use some sort of heuristic / parser to find the relevant bits using the brackets as guides for how to find it :D

Maybe it's just build a tree where everything inside a backet of any sorts becomes a sub-tree... then look for a node with the label `defaultConfig`, find children named `minSdkVersion` look for the next integer.. or something like that.